journalpump/senders/rsyslog.py

```diff
@@ -56,6 +56,10 @@ def _init_rsyslog_client(self):
     def send_messages(self, *, messages, cursor):
         if not self.rsyslog_client:
             self._init_rsyslog_client()
+
+        if self.rsyslog_client is None:
+            raise Exception("failed to initialize rsyslog client")
```

Comment:
I don't think it is the purpose of this function to ensure that the previous function was succesful. I'd consider moving this failure

Reply:
See https://github.com/aiven/journalpump/pull/72#discussion_r475588369

The only reasonable change with the current structure is to replace it for an assert .